File: LeetCode/Python/Medium/1167-Minimum-Cost-to-Connect-Sticks/solution.py
import heapq

class Solution:
    def connectSticks(self, sticks: List[int]) -> int:
        result=0
        
        min_heap=list()

        for stick in sticks:
            heapq.heappush(min_heap, stick)

        while len(min_heap) > 1:
            x=heapq.heappop(min_heap)

            y=heapq.heappop(min_heap)

            total=x + y

            result+=total

            heapq.heappush(min_heap, total)
            
        return result

File: LeetCode/Rust/Medium/1167-Minimum-Cost-to-Connect-Sticks/solution.rs
use std::collections::{BinaryHeap};
use std::cmp;

impl Solution
{
    pub fn connect_sticks(sticks: Vec<i32>) -> i32
    {
        let mut result: i32=0;

        let mut min_heap: BinaryHeap<cmp::Reverse<i32>>=BinaryHeap::new();

        for &stick in sticks.iter()
        {
            min_heap.push(cmp::Reverse(stick));
        }

        while min_heap.len() > 1
        {
            let x: i32=min_heap.pop().unwrap().0;

            let y: i32=min_heap.pop().unwrap().0;

            let total: i32=x + y;

            result+=total;

            min_heap.push(cmp::Reverse(total));
        }

        result
    }
}